recipes/newrelic/apm/.net/linux/net-debian.yml

```diff
@@ -0,0 +1,76 @@
+# Visit our schema definition for additional information on this file format
+# https://github.com/newrelic/open-install-library/blob/main/docs/recipe-spec/recipe-spec.md#schema-definition
+
+name: dotnet-agent-installer
+displayName: .NET Agent Installer
+description:  New Relic install recipe for .NET application on Linux
+repository: https://github.com/newrelic/newrelic-dotnet-agent
+
+installTargets:
+  - type: host
+    os: linux
+    platform: debian
+
+keywords:
+  - Apm
+  - .NET
+  - dotnet
+  - aspnet
+  - core
+
+validationNrql: "SELECT count(*) from Transaction WHERE host like '{{.HOSTNAME}}%' facet entityGuid since 10 minutes ago"
+
+install:
+
+  version: "3"
+  silent: true
+
+  tasks:
+    default:
+      cmds:
+        - task: assert_pre_req
+        - task: remove_any_previous
+        - task: install
+        - task: configure
+
+    assert_pre_req:
+      cmds:
+        - |
+          results=$(dpkg-query -W -f='${Status}\n' dotnet-runtime-*)
+          if [ $? -ne 0 ]; then
+            # Did not find the dotnet-runtime package, don't install the agent
+            exit 1
+          fi
+
+    remove_any_previous:
+      cmds:
+        - sudo apt-get purge -y newrelic-netcore20-agent
+        - |
+          AGENTPROFILE=/etc/profile.d/newrelic-netcore20-agent-path.sh
+          if test -f "$AGENTPROFILE"; then
+              sudo rm $AGENTPROFILE
+          fi
+
+    install:
+      cmds:
+        - echo 'deb http://apt.newrelic.com/debian/ newrelic non-free' | sudo tee /etc/apt/sources.list.d/newrelic.list
+        - wget -O- https://download.newrelic.com/548C16BF.gpg | sudo apt-key add -
+        - sudo apt-get update
+        - sudo apt-get install newrelic-netcore20-agent
+        - echo "New Relic .NET agent installed"
+
+    configure:
+      cmds:
+        - |
+          echo "Setup startup environment variables"
+          echo 'export NEW_RELIC_LICENSE_KEY={{.NEW_RELIC_LICENSE_KEY}}' | sudo tee -a /etc/profile.d/newrelic-netcore20-agent-path.sh
+          
+          echo "Setup session environment variables for child processes"
+          export NEW_RELIC_LICENSE_KEY={{.NEW_RELIC_LICENSE_KEY}}
```

Comment:
I think those export need to be written into the ~/.bashrc file, so they persist on the next session. You can see an example here https://github.com/newrelic/open-install-library/blob/main/test/deploy/linux/haproxy/install/debian/roles/configure/tasks/main.yml#L36

Reply:
I was not sure we wanted to write those out to a place that loads them all be default.  We don't recommend enabling the agent to instrument all .NET Core apps globally. We do this due to how the Core parts of the profiler are designed.  On the Framework side, we require some configuration changes to tell the profiler to explicitly instrument apps outside of IIS.  The profiler loads briefly, checks if it should be there, and if not it unloads.  For .NET Core, this was never implemented - to keep it simple - so the profiler will stay loaded, watching methods, but never instrumenting them.

Reply:
Ho, I don't quite fully understand, but here is what I know.
Those export would be gone the moment the recipe install is completed (like they were not issued).
To make those persistent those should be in the ~/.bashrc. And also, just noticed this morning, if this is done, it should be idempotent (only written at most once in the file).
I'll check if the .net agent instrument the demo apps, and will get back to you.